ui/src/projects.vue

```diff
@@ -221,6 +226,37 @@ export default {
                 alert('Please signup/login first to create a new project');
             }
         },
+    },
+    computed: {
+        userAgreementAgreed: function() {
+            if(Vue.config.user && Vue.config.profile) {
+                const latestAgreementDate = new Date('2023-08-17'); // Replace with the actual date of the latest version
+                const userAgreementDateStr = Vue.config.profile.private.aup;
+                console.log("User agreement date", userAgreementDateStr);
+
+                if (typeof userAgreementDateStr === "boolean") {
+                    console.log("User agreement date is a boolean value:", userAgreementDateStr);
+                    return false;
+                }
+
+                const userAgreementDate = new Date(userAgreementDateStr);
+
+                if (!(userAgreementDate instanceof Date && !isNaN(userAgreementDate.getTime()))) {
+                    console.log("User agreement date is not a valid date", userAgreementDate);
+                    return false;
+                } 
+
+                // User has not accepted the latest version of the data use agreement
+                if (userAgreementDate < latestAgreementDate) {
+                    console.log("User has not accepted the latest version of the data use agreement", userAgreementDate, latestAgreementDate);
+                    return false;
+                }
+
+                return true;
+            }
+           
+            return true;
```

Comment:
shouldnt this be return false?

Reply:
brainlife projects page can be accessed without logging in so it handles that case

Reply:
ok I'd suggest two things:
- move the `userAgreementAgreed` logic to inside the modal component, to keep it in one place, and show the modal on mounted if the logic is true
- ` v-if=` the modal to check if the user is logged. calling it `userAgreementAgreed` and returning true for non-logged users is misleading

Reply:
yeah! thank you